### ISSUE TITLE
Voice Activity detection using DeepSpeech

### ISSUE THREAD

**Issue by ankitmundada** [SeanNaren/deepspeech.pytorch#266]
This is not an issue but a general question. 
I would like to explore the possibility of using DeepSpeech as a `Voice Activity Detector` giving exact timestamps for `voice` and `silence` in an audio clip. Has anyone tried doing this? What will be the best way to build such a model (even if it is not using DeepSpeech)?

**Comment by ryanleary** [comment]
There are a number of approaches for this in the literature. You almost certainly _could_ rework this model to do that, but my initial intuition is that it is tremendously overkill for that problem unless you _also_ intend on doing STT.

**Comment by ankitmundada** [comment]
@ryanleary Thank you for your reply.
I'm interested in using this model, because:
- We're using it extensively in our system and it'd be much easier to make some changes and get it working for `VAD`
- I'll definitely have a look at other approaches in literature, but before that, I'd like to explore the possibility of reworking this model as I'm pretty familiar with it now. 

I am considering the following approach to customize this model, please let me know of what you think:
- The targets will have a label for each fixed time-window (Ex. if the file is 1sec long, with 0.5 sec voice and 0.5 sec silence, the labels will be `VS` (V-voice, S-Silence), considering each label corresponds to a 0.5 sec segment (time-window) of the input clip)
- Instead of using `CTCLoss`, using `CrossEntropyLoss` over the softmax outputs

You're most welcome to make suggestions :)

**Comment by WillQuCD** [comment]
@ankitmundada you can try to change labels to vs. It should be work.

**Comment by harrypotter90** [comment]
Rhhjhhjnijkimo213456jkjjknnmkkjnnknjbvcxzbnnbbñnn nknnmnnbbbv nnmnb   njjjk

**Comment by Vintic** [comment]
Some updates on that issue?
